src/zcash/prf.cpp

```diff
@@ -0,0 +1,63 @@
+#include "prf.h"
+#include "crypto/sha256.h"
+
+uint256 PRF(bool a, bool b, bool c, bool d,
+            const uint256& x,
+            const uint256& y)
+{
+    uint256 res;
+    unsigned char blob[64];
+
+    memcpy(&blob[0], x.begin(), 32);
+    memcpy(&blob[32], y.begin(), 32);
+
+    blob[0] &= 0x0F;
+    blob[0] |= (a ? 1 << 7 : 0) | (b ? 1 << 6 : 0) | (c ? 1 << 5 : 0) | (d ? 1 << 4 : 0);
```

Comment:
Personally, I'd rather have a separate ticket addressing this. It's good for security *analysis*, but during this transitional period of the code, it's an annoying invariant to enforce, and far easier to test and review later when the API is simpler. If you agree can you make a ticket about it?

(Another reason is that I would like to make a kind of `uint252` type which centralizes these invariants. This also becomes much easier to do later when the API is cleaner.)